src/dispatch/search/service.py

```diff
@@ -19,12 +21,13 @@ def get_by_name(*, db_session, name: str) -> Optional[SearchFilter]:
     return db_session.query(SearchFilter).filter(SearchFilter.name == name).first()
 
 
-def match(*, db_session, filter_spec, model, model_id: int):
-    """Matches an incident with a given search filter."""
-    model_cls = get_class_by_tablename(model)
+def match(*, db_session, filter_spec: List[dict], class_instance: Base):
+    """Matches an class instance with a given search filter."""
+    table_name = get_table_name_by_class_instance(class_instance)
+    model_cls = get_class_by_tablename(table_name)
```

Comment:
Why not simply pass the model directly? e.g `Incident` from `dispatch.incident.models`, the caller should know what they are trying to match with?

Reply:
You would have to change the signature, though to something like: `def match(*, db_session, filter_spec: List[dict], model: Any, instance: Any):` or similar.  I feel as though all of the table naming stuff gets a bit dicy if for some reason some table is named weirdly.

Reply:
I wanted to minimize the amount of parameters needed when calling the `notification.send()` function. Function `get_table_name_by_class_instance()` should be pretty reliable. We can adjust in the future if it causes issues.